analyses/mwi/dryspells_trigger/exploration/ecmwf_highres_compare_aggr.md

```diff
@@ -0,0 +1,365 @@
+    sel_lt=3
+    num_plots=len(meth_comb)
+    rows = math.ceil(num_plots / colp_num)
+    position = range(1, num_plots + 1)
+    fig=plt.figure(figsize=figsize)
+    df_comb_plot=df_comb_pivot.copy()
+    df_comb_plot_sel=df_comb_plot[df_comb_plot.leadtime==sel_lt]
+
+    for i, meth in enumerate(meth_comb):
+        ax = fig.add_subplot(rows,colp_num,i+1)
+        val_col=f"{meth[0]}-{meth[1]}"
+        df_comb_plot_sel[val_col]=df_comb_plot_sel[meth[0]]-df_comb_plot_sel[meth[1]]
+        df_comb_plot_sel.sort_values(val_col,inplace=True,ascending=False)
+        df_comb_plot_sel["colors"]=np.where(df_comb_plot_sel["dry_spell"]==1,ds_color,no_ds_color)
+        df_comb_plot_sel=df_comb_plot_sel.reset_index(drop=True)
+        # Plotting the horizontal lines
+        plt.hlines(y=df_comb_plot_sel.index
+                , xmin=0, xmax=df_comb_plot_sel[val_col],
+                   color=df_comb_plot_sel.colors,
+                   linewidth=5)
+
+        # Decorations
+        # Setting the labels of x-axis and y-axis
+        plt.gca().set(ylabel='Month', xlabel=f'Difference (mm),{val_col.replace("-"," minus ")}')
+
+        # Setting Date to y-axis
+        plt.yticks(df_comb_plot_sel.index, df_comb_plot_sel.date.dt.strftime("%Y-%m"), fontsize=12)
+        ax.xaxis.label.set_size(16)
+
+        plt.title(f'{val_col.replace("-"," minus ")}', fontdict={
+                  'size': 20})
+        plt.grid(linestyle='--', alpha=0.5)
+        ds_patch = mpatches.Patch(color=ds_color, label="dry spell")
+        nods_patch = mpatches.Patch(color=no_ds_color, label="no dry spell")
+
+        ax.set_xlim(-20,55)
+#         ax.set_xlim(-40,35)
+
+        plt.legend(handles=[ds_patch, nods_patch])
+    fig.suptitle("Difference in forecasted precipitation at leadtime=3 for different aggregation methods",fontsize=30)
+    fig.tight_layout(rect=[0, 0.03, 1, 0.98])
+```
+
+```python
+meth_comb=[[m,"cds_unrounded_center"] for m in ["api_center","api_weightavg","api_alltouched"]]
+plt_div_bar(meth_comb,colp_num=3)
+```
+
+#### Boxplot
+Only 20 datapoints per boxplot, so boxplot might not be the best method
+
+```python
+plot=alt.Chart().mark_boxplot(size=50).encode(
+    x=alt.X('method:N',sort=df_comb.method.unique()),
+    y=alt.Y(stat_col_for,scale=alt.Scale(zero=False),title="Forecasted precipitation"),
+#     facet=alt.Facet(column="month_name:N",sort=df_comb.month_name.unique())
+).properties(width=400,height=400)
+plot.facet(column=alt.Column("month_name:N",sort=df_comb.month_name.unique(),title="Month"),row=alt.Row("leadtime:N",sort=[2,3,4]),data=df_comb[df_comb.leadtime.isin([2,3,4])],
+           title=["Distribution of forecasted precipitation for different methods, facetted by leadtime and month", "Each boxplot only contains 20 datapoints"])
+```
+
+#### Histogram
+Stacked based on occurrence of dry spell    
+Due to so little observations it is not really a distrubtion which makes it hard to compare
```

Comment:
There are statistical tests you can use if you're interested in knowing whether the data could come from the same underlying distribution (like the KS test)

Reply:
Hmm that would maybe have been a good idea to compare the full distribution between methods. However I would say splitting between dry spells and non-dry spells we would have too little data to even estimate a distribution? 

I think for now since we rounded off this exploration I will not make that addition. Is that okay?

Reply:
Of course! Just putting the idea out there.